lib/winapi/timezoneapi.c

```diff
@@ -0,0 +1,197 @@
+    }
+
+    // Xbox stores up to 4 characters here
+    for (UCHAR i = 0; i < 4; i++) {
+        lpTimeZoneInformation->StandardName[i] = standardName[i];
+        lpTimeZoneInformation->DaylightName[i] = dayLightName[i];
+    }
+
+    // There was no DST info
+    if (dayLightEndDate.Month == 0 || dayLightStartDate.Month == 0) {
+        return TIME_ZONE_ID_UNKNOWN;
+    }
+
+    lpTimeZoneInformation->StandardDate.wMonth = dayLightEndDate.Month;
+    lpTimeZoneInformation->StandardDate.wDayOfWeek = dayLightEndDate.DayOfWeek;
+    lpTimeZoneInformation->StandardDate.wDay = dayLightEndDate.Day;
+    lpTimeZoneInformation->StandardDate.wHour = dayLightEndDate.Hour;
+    lpTimeZoneInformation->StandardBias = standardBias;
+
+    lpTimeZoneInformation->DaylightDate.wMonth = dayLightStartDate.Month;
+    lpTimeZoneInformation->DaylightDate.wDayOfWeek = dayLightStartDate.DayOfWeek;
+    lpTimeZoneInformation->DaylightDate.wDay = dayLightStartDate.Day;
+    lpTimeZoneInformation->DaylightDate.wHour = dayLightStartDate.Hour;
+    lpTimeZoneInformation->DaylightBias = dayLightBias;
+
+    // Now we need to determine if the current time is within the DST range
+    KeQuerySystemTime(&kTime);
+    RtlTimeToTimeFields(&kTime, &dateNow);
+
+    // Determine what day in the month DST starts and ends
+    UCHAR dayLightStartDay = GetActualDay(dateNow.Year, &dayLightStartDate);
+    UCHAR dayLightEndDay = GetActualDay(dateNow.Year, &dayLightEndDate);
+
+    // Fix month wrapping. i.e if DST is from Oct(10) to Feb(2) and the current month is Jan(1) we should be in DST
+    // Applying a simple check would result is 10 < 1 < 2 = false and incorrect
+    // To account for this, we offset the months in the following year by 12
+    // This results in the comparison 10 < 13(12+1) < 14(12+2) = true and is correct
+    if (dayLightEndDate.Month < dayLightStartDate.Month)
+    {
+        if (dateNow.Month <= dayLightEndDate.Month) {
+            dateNow.Month += 12;
+        }
+        dayLightEndDate.Month += 12;
+    }
+
+    // Now that month wrapping is fixed, we convert the dates to minutes from the start of the year corresponding to dayLightStartDate
+    // The conversion to minutes allows for easier comparison. The minute resolution allows for timezones with 0.5 hour offsets.
+    const LONG MPD = 24 * 60;
+    const LONG MPM = 31 * MPD; //  31 is ok providing it's consistent
+    timeNow = (dateNow.Month * MPM) + (dateNow.Day * MPD) + (dateNow.Hour * 60) + dateNow.Minute;
+    // DST change overs happen in local time, so we need to add the respective biases too
+    dayLightStart = (dayLightStartDate.Month * MPM) + (dayLightStartDay * MPD) + (dayLightStartDate.Hour * 60) + timeZoneBias + standardBias;
+    dayLightEnd = (dayLightEndDate.Month * MPM) + (dayLightEndDay * MPD) + (dayLightEndDate.Hour * 60) + timeZoneBias + dayLightBias;
+
+    if (timeNow >= dayLightStart && timeNow < dayLightEnd) {
+        return TIME_ZONE_ID_DAYLIGHT;
+    }
+    else {
```

Comment:
nit: Formatting in other files (also within the nxdk winapi) has `} else {`. Not sure about winapi or if there's a code convention for nxdk now (clang-format?)

Reply:
missed this formatting issue. Done